src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11041,12 +11101,25 @@ private LambdaExpressionSyntax ParseLambdaExpression(SyntaxToken asyncToken)
             else
             {
                 var name = this.ParseIdentifierToken();
-                var arrow = this.EatToken(SyntaxKind.EqualsGreaterThanToken);
-                arrow = CheckFeatureAvailability(arrow, MessageID.IDS_FeatureLambda);
-
+                // PROTOTYPE: TryEatToken
+                var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
+                SyntaxToken arrow;
+                if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken && this.PeekToken(1).Kind == SyntaxKind.GreaterThanToken)
+                {
+                    var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
```

Comment:
unlike in the higher up parsing code, you now don't assign anything to `exclamation` even though you see the user wrote `x!=>`.  We should.  Note: this is why we should also have separate branches for all these cases.  because you don't want to overwrite the assignment that happened above.